src/input/keyboard/mod.rs

```diff
@@ -918,6 +921,14 @@ impl<D: SeatHandler + 'static> KeyboardHandle<D> {
         }
     }
 
+    /// Calls `f` with the active grab, if any.
+    pub fn with_grab(&self, f: impl FnOnce(Serial, &dyn KeyboardGrab<D>)) {
+        let guard = self.arc.internal.lock().unwrap();
+        if let GrabStatus::Active(s, g) = &guard.grab {
+            f(*s, &**g)
+        }
+    }
```

Comment:
From the example it might make sense to have the function return something maybe?

Something like:

```rs
    pub fn with_grab<R>(&self, f: impl FnOnce(Serial, &dyn KeyboardGrab<D>) -> Option<R>) {
        let guard = self.arc.internal.lock().unwrap();
        if let GrabStatus::Active(s, g) = &guard.grab {
            Some(f(*s, &**g))
        } else {
           None
        }
    }
```

Could then make the example a bit more convenient: 

```rs
let mut is_dnd_grab = 
pointer.with_grab(|_, grab| {
    grab.as_any().downcast_ref::<DnDGrab<Self>>().is_some()
}).unwrap_or(false);
```

Reply:
I was debating between this and the simpler one in the PR, kind of 50/50 between them.

Reply:
Afaik most function in smithay that operate on a closure allow to return something, so I would prefer we add it here too.

Reply:
Did that